rewrite-core/src/main/java/org/openrewrite/internal/ListUtils.java

```diff
@@ -188,8 +193,17 @@ public static <T> List<T> flatMap(@Nullable List<T> ls, BiFunction<Integer, T, O
         }
 
         if (newLs != ls) {
-            //noinspection StatementWithEmptyBody
-            while (newLs.remove(null)) ;
+            boolean identical = newLs.size() == ls.size();
+            for (int i = newLs.size() - 1; i >= 0; i--) {
+                T newElem = newLs.get(i);
+                identical = identical && newElem == ls.get(i);
```

Comment:
if `newLs` has a bigger size than `ls`, then `ls.get(i)` will be out of the boundary.

Reply:
If the sizes are not the same, `identical` will be initialized as `false` and the short-circuiting property of `&&` makes sure the RHS never gets evaluated then. But let me add a test for that.